pypdf/generic/_base.py

```diff
@@ -409,8 +412,8 @@ def clone(
     def myrepr(self) -> str:
         if self == 0:
             return "0.0"
-        nb = int(log10(abs(self)))
-        s = f"{self:.{max(1,16-nb)}f}".rstrip("0").rstrip(".")
+        nb = FLOAT_WRITE_PRECISION - int(log10(abs(self)))
```

Comment:
I need some notes to wrap my head around this:

Before the change, `nb` was essentially just the power of 10:

...
* (0.01, 0.1]: `nb=-1` leads to `.17f`
* (0.1, 9): `nb=0` leads to `.16f`
* [10, 99): `nb=1` leads to `.15f`
...

The reasoning was that if we something closer to 0, we need more precision. The bigger the number gets, the less the digits after the decimal point matters.

With the change:

...
* (0.01, 0.1]: `nb=9` leads to `.9f`
* (0.1, 9): `nb=8` leads to `.8f`
* [10, 99): `nb=7` leads to `.7f`
...

Ok, still the same logic. 

edit: oh, got it. I think there is a simpler change

Reply:
before the change FLOAT_WRITE_PRECISION was set to 16 which was too high to be compatible all mantises values